src/rust/engine/src/intrinsics.rs

```diff
@@ -106,7 +113,13 @@ impl Intrinsics {
       Box::new(run_id),
     );
     intrinsics.insert(
-      Intrinsic::new(types.interactive_process_result, types.interactive_process),
+      Intrinsic {
+        product: types.interactive_process_result,
+        inputs: vec![
+          DependencyKey::new(types.interactive_process),
+          DependencyKey::new(types.process_config_from_environment),
+        ],
+      },
       Box::new(interactive_process),
     );
```

Comment:
While this makes sense from the perspective of symmetry, an `InteractiveProcess` is not going to be able to run inside of docker anytime soon. There would be a lot to figure out for that, and it doesn't seem like a first-version feature (cc @tdyas).

But one reasonable way to do that would probably be to do what you've done here, but then validate in `fn interactive_process` below that a docker image has not been set, and fail with an error if so ("Docker is not yet supported for interactive processes").